### PR TITLE
Create integration-Twilio.yml

### PR DESCRIPTION
**Pull request by shaniacht1**

### CODE REVIEW

File: Integrations/integration-Twilio.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  type: 8
+  required: false
+- display: Use system proxy settings
+  name: proxy
+  defaultvalue: ""
+  type: 8
+  required: false
+script:
+  script: |2+
+
```
Comment (Spivaka) [inline Integrations/integration-Twilio.yml]: remove those spaces

File: Integrations/integration-Twilio.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                    Body: body,
+                    Headers: headers,
+                    Username: params.sid,
+                    Password: params.token
+                },
+                params.insecure,
+                params.proxy
+            );
+            if (res.StatusCode < 200 || res.StatusCode >=300) {
+                throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
```
Comment (Spivaka) [inline Integrations/integration-Twilio.yml]: You are not stringifying the body here... and not sure it is json... just print res.Body

File: Integrations/integration-Twilio.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            }
+            return res.Body;
+        };
+
+
+
+
+    var headers = {'Content-Type': ['application/x-www-form-urlencoded']};
```
Comment (Spivaka) [inline Integrations/integration-Twilio.yml]: Don't pass headers as a param if it constant, use it in line 58 directly

File: Integrations/integration-Twilio.yml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  defaultvalue: ""
+  type: 8
+  required: false
+script:
+  script: |2+
+
+
+
+    var sendRequest = function(url, body, headers) {
+            url.replace(/[\/]+$/, '');
```
Comment (Spivaka) [inline Integrations/integration-Twilio.yml]: no need for this, you are cleaning the server url outside

File: Integrations/integration-Twilio.yml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+
+
+    var headers = {'Content-Type': ['application/x-www-form-urlencoded']};
+    var url = params.server.replace(/[\/]+$/, '')+'/'+params.sid+'/Messages.json' ;
+    // The command input arg holds the command sent from the user.
+    switch (command) {
+        // This is the call made when pressing the integration test button.
+        case 'test-module':
+            var body = "From=%2B"+params.sender.replace(/^\+/, '')+"&To=%2B"+params.sender.replace(/^\+/, '')+"&Body="+args.body;
```
Comment (Spivaka) [inline Integrations/integration-Twilio.yml]: you should url encode the body using encodeURIComponent(args.body) before sending it

File: Integrations/integration-Twilio.yml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+                throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+            }
+            return res.Body;
+        };
+
+
+
+
+    var headers = {'Content-Type': ['application/x-www-form-urlencoded']};
+    var url = params.server.replace(/[\/]+$/, '')+'/'+params.sid+'/Messages.json' ;
```
Comment (Spivaka) [inline Integrations/integration-Twilio.yml]: URL is also a const, built it inside the send request than

File: Integrations/integration-Twilio.yml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+
+  type: javascript
+  commands:
+  - name: Twilio_sendSMS
+    arguments:
+    - name: reciever
+      required: true
+      description: Phone number of SMS reciever
+    - name: body
```
Comment (Spivaka) [inline Integrations/integration-Twilio.yml]: is this name backward compatible?

File: Integrations/integration-Twilio.yml [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        default:
+           var body = "From=%2B"+params.sender.replace(/^\+/, '')+"&To=%2B"+args.reciever.replace(/^\+/, '')+"&Body="+args.body;
+           sendRequest(url, body, headers);
+           return 'Message sent successfully!';
+    }
+
+
+  type: javascript
+  commands:
+  - name: Twilio_sendSMS
```
Comment (Spivaka) [inline Integrations/integration-Twilio.yml]: If you use casing, don't also use _
Should be
TwilioSendSMS
or
twilio_send_sms
not a mixture

**Comment by aviadl** [comment]
@shaniacht1 @Spivaka what is the status of this PR ?

**Comment by aviadl** [comment]
This needs to be merged ASAP, else it will cause upgrade problems to existing customers, since the old code was already deleted from the server

**Pull request merged**